### PR TITLE
feat: kafka plugin port

### PR DESCRIPTION
**Pull request by algchoo**
### Proposed Change

Ported the Apache Kafka Stanza plugin over and tested against logs from a live system.

**config.yaml**
```
receivers:
  plugin:
    path: ./plugins/kafka_logs.yaml
    parameters:
      enable_server_log: true
      server_log_path:
        - /Users/austingeorgiades/kafka-logs/logs/server.log*
      enable_controller_log: true
      controller_log_path:
        - /Users/austingeorgiades/kafka-logs/logs/controller.log*
      enable_state_change_log: true
      state_change_log_path:
        - /Users/austingeorgiades/kafka-logs/logs/state-change.log*
      enable_log_cleaner_log: true
      log_cleaner_log_path:
        - /Users/austingeorgiades/kafka-logs/logs/log-cleaner.log*
      start_at: beginning

exporters: 
  logging:
    logLevel: debug

service:
  pipelines:
    logs:
      receivers:
      - plugin
      exporters:
      - logging
```

**test data:**
[logs.zip](https://github.com/observIQ/observiq-otel-collector/files/8741701/logs.zip)

**sample output:**
```
2022-05-20T11:59:51.921-0400    DEBUG   loggingexporter/logging_exporter.go:81  ResourceLog #0
Resource SchemaURL: 
ScopeLogs #0
ScopeLogs SchemaURL: 
InstrumentationScope  
LogRecord #0
ObservedTimestamp: 2022-05-20 15:59:51.821911 +0000 UTC
Timestamp: 2022-05-18 17:56:59.593 +0000 UTC
Severity: Trace
Body: [2022-05-18 17:56:59,593] TRACE [Controller id=0] Leader imbalance ratio for broker 0 is 0.0 (kafka.controller.KafkaController)
Attributes:
     -> message: STRING([Controller id=0] Leader imbalance ratio for broker 0 is 0.0 (kafka.controller.KafkaController))
     -> log_type: STRING(kafka.controller)
     -> log.file.name: STRING(controller.log.2022-05-18-17)
     -> time: STRING(2022-05-18 17:56:59,593)
     -> severity: STRING(TRACE)
Trace ID: 
Span ID: 
Flags: 0
LogRecord #1
ObservedTimestamp: 2022-05-20 15:59:51.821937 +0000 UTC
Timestamp: 2022-05-18 18:30:59.085 +0000 UTC
Severity: Info
Body: [2022-05-18 18:30:59,085] INFO [Controller id=0] Resigned (kafka.controller.KafkaController)
Attributes:
     -> log.file.name: STRING(controller.log)
     -> time: STRING(2022-05-18 18:30:59,085)
     -> severity: STRING(INFO)
     -> message: STRING([Controller id=0] Resigned (kafka.controller.KafkaController))
     -> log_type: STRING(kafka.controller)
Trace ID: 
Span ID: 
Flags: 0
```

##### Checklist
- [x] Changes are tested
- [ ] CI has passed

### CODE REVIEW

File: plugins/kafka_logs.yaml [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+          - filelog/controller
+          {{ end }}
+
+          {{ if .enable_state_change_log }}
+          - filelog/state
+          {{ end }}
+
+          {{ if .enable_log_cleaner_log }}
+          - filelog/cleaner
+          {{ end }}
```
Comment (BinaryFissionGames) [inline plugins/kafka_logs.yaml:187]: newline

**Review by BinaryFissionGames — APPROVED** [review]
This looks good to me, just needs a newline at the end

File: plugins/kafka_logs.yaml (hunk 1 above)
Comment (cpheps) [reply plugins/kafka_logs.yaml:187]: @algchoo If you can fix this newline I'll approve.

**Review by cpheps — APPROVED** [review]

File: plugins/kafka_logs.yaml (hunk 1 above)
Comment (algchoo) [reply plugins/kafka_logs.yaml]: @cpheps fixed it, thanks!

**Pull request merged**